Detectors/EMCAL/simulation/src/LZEROElectronics.cxx

```diff
@@ -0,0 +1,241 @@
+          first = false;
+        } else {
+          // summedDigit += ld;
+
+          // safety device in case same fastOr
+          // but different towers, i.e. remember
+          // that the += operator fails WITHOUT
+          // feedback if that were to happen
+          Digit digitToSum(summedDigit.getTower(), ld.getAmplitude(), summedDigit.getTimeStamp());
+          summedDigit += digitToSum;
+        }
+      }
+
+      sizemDigitMap = (*digitsTimeBin.mDigitMap).size();
+      if (mSimulateNoiseDigits) {
+        addNoiseDigits(summedDigit);
+      }
+
+      if (summedDigit.getAmplitude() < 0) {
+        continue;
+      }
+
+      auto whichTRU = std::get<0>(mTriggerMap->getTRUFromAbsFastORIndex(fastor));
+      auto whichFastOrTRU = std::get<1>(mTriggerMap->getTRUFromAbsFastORIndex(fastor));
+
+      auto whichFastOr = std::get<1>(mTriggerMap->convertFastORIndexTRUtoSTU(whichTRU, whichFastOrTRU));
+      auto& patchTRU = patchesFromAllTRUs[whichTRU];
+      auto& fastOrPatchTRU = patchTRU.mFastOrs[whichFastOr];
+      fastOrPatchTRU.updateADC(summedDigit.getAmplitudeADC());
+
+      digIndex++;
+      counterhelp++;
+    }
+
+    // Evaluate -> peak finder (ALL Patches in ALL TRUs)
+    // in case peak found:
+    // - Create trigger input (IR of that timebin - delay [typically 8 or 9 samples - rollback (0)])
+    // - Create L1 timesums (trivial - last time integral) -> Collect from all fastOrs in ALL TRUs
+
+    // Trigger Inputs needs to have the correction of the delay
+    // Propagating for now all the interaction record
+    // The typical delay is 8 to 9 BCs and the rollback as well
+    // The rollback is taken from EMCALReconstruction/RecoParam.h
+    // The delay is due to the interactions between EMCAL and CTP
+    // It accounts for the difference in times between L0a, L0b, and then there will be a L1 and L1b delay
+    // There is 1BC uncertainty on the trigger readout due to steps in the interaction between CTP and detector simulations
+    bool foundPeak = false;
+    for (auto& patches : patchesFromAllTRUs) {
+      updatePatchesADC(patches);
+      bool foundPeakCurrentTRU = peakFinderOnAllPatches(patches);
+      auto firedPatches = getFiredPatches(patches);
+      if (foundPeakCurrentTRU)
+        foundPeak = true;
+    }
+
+    if (foundPeak == true)
+      LOG(debug) << "DIG TRU fill in LZEROElectronics: foundPeak = " << foundPeak;
+    EMCALTriggerInputs TriggerInputsForL1;
+    if (foundPeak) {
+      TriggerInputsForL1.mInterRecord = record;
+      int whichTRU = 0;
+      for (auto& patches : patchesFromAllTRUs) {
+        if (whichTRU < 52) {
+          int whichFastOr = 0;
+          for (auto& fastor : patches.mFastOrs) {
+            TriggerInputsForL1.mLastTimesumAllFastOrs.push_back(std::make_tuple(whichTRU, std::get<1>(mTriggerMap->convertFastORIndexSTUtoTRU(mTriggerMap->convertTRUIndexTRUtoSTU(whichTRU), whichFastOr, o2::emcal::TriggerMappingV2::DetType_t::DET_EMCAL)), fastor.timesum()));
```

Comment:
`o2::emcal::TriggerMappingV2::DetType_t` is there to distinguish between EMCAL and DCAL STU. Currently you put everything under EMCAL. You need to distinguish the STU based on the TRU index.

Reply:
These files are only needed for debug purposes and to produce correlation plots. Only the TRU index is needed. However, the algorithm considers the geometry of the TRU i.e. if it is full, third, C or A side.